Implement sshfs volumes for podman machine (#12584)
containers/podman: docs/source/markdown/podman-machine-init.1.md

Start the virtual machine immediately after it has been initialized.

#### **--volume**, **-v**=*source:target*

Mounts a volume from source to target.

> Create a mount. If you specify /host-dir:/machine-dir, Podman mounts _host-dir_ in the host to _machine-dir_ in the Podman machine.

Reviewer: As a practice, we avoid pronouns in the man pages:
```suggestion
Create a mount. If /host-dir:/machine-dir is specified as the `*source:target*`, Podman mounts _host-dir_ in the host to _machine-dir_ in the Podman machine.
```

Author: Will change these in the other PR

> In the default operating system, the root filesystem is mounted read-only, so mounts need to be created under e.g. /mnt directory.

Reviewer: ```suggestion
The root filesystem is mounted read-only in the default operating system, so mounts must be created under the /mnt directory.
```

Author: Will change these in the other PR

> #### **--volume-driver**

Reviewer: It would be nice to have an example of this in the Example section too.

Author: Will change these in the other PR
